src/SnoopCompile.jl

```diff
@@ -25,29 +27,32 @@ function snoop(filename, commands)
     # addprocs will run the unmodified version of julia, so we
     # launch it as a command.
     code_object = """
-        while !eof(STDIN)
-            eval(Main, deserialize(STDIN))
-        end
-        """
-    in, io = open(`$(Base.julia_cmd()) --eval $code_object`, "w", STDOUT)
-    serialize(in, quote
-        import SnoopCompile
+            using Serialization
+            using Pkg
+            Pkg.activate(abspath("."))
+            while !eof(stdin)
+                Core.eval(Main, deserialize(stdin))
+            end
+            """
+    in_ = open(`$(Base.julia_cmd()) --eval $code_object`, stdout, write=true)
+    serialize(in_, quote
+              import SnoopCompile
     end)
     # Now that the new process knows about SnoopCompile, it can
     # expand the macro in this next expression
-    serialize(in, quote
+    serialize(in_, quote
           SnoopCompile.@snoop1 $filename $commands
     end)
-    close(in)
-    wait(io)
+    # close(in_)
+    # wait(in_)
     println("done.")
     nothing
 end
 
 function split2(str, on)
-    i = search(str, on)
+    i = something(findfirst(isequal(on), str), 0)
```

Comment:
The more Julian way now would be to just look for `nothing` as the output of `findfirst`. (It's less ambiguous than using a 0 index as "not found" since arrays can now have 0 and negative indices.)